src/definitions/modules/dropdown.js

```diff
@@ -366,11 +365,22 @@ $.fn.dropdown = function(parameters) {
             }
             if( module.is.search() && !module.has.search() ) {
               module.verbose('Adding search input');
-              $search = $('<input />')
+              if($module.prev('label').length) {
+                $search = $('<input />')
                 .addClass(className.search)
                 .prop('autocomplete', module.is.chrome() ? 'fomantic-search' : 'off')
+                .attr('aria-labelledby',id+'_formLabel')
                 .insertBefore($text)
-              ;
+                ;
+                $module.prev('label').attr('id', id + '_formLabel');
+              }
+              else {
```

Comment:
Let's not duplicate the code here.
I suggest to change your PR into something like this (which will also support an already existing id as [questioned by @hdogan](https://github.com/fomantic/Fomantic-UI/pull/2409/files#r928965797) )
```javascript
var labelNode = $module.prev('label');
$search = $('<input />')
  .addClass(className.search)
  .prop('autocomplete', module.is.chrome() ? 'fomantic-search' : 'off')
;
if (labelNode.length) {
  if(!labelNode.attr('id')) {
    labelNode.attr('id', module.get.id() + '_formLabel');
  }
  $search.attr('aria-labelledby',labelNode.attr('id'));
}
$search.insertBefore($text);
```

Reply:
thank you so much, it is a lot better now